cmd/cerbos/repl/internal/repl.go

```diff
@@ -688,6 +694,137 @@ func (r *REPL) mkEnv() (*cel.Env, error) {
 	return conditions.StdEnv.Extend(cel.Declarations(decls...))
 }
 
+func (r *REPL) Complete(line string) []string {
+	parts := strings.Fields(line)
+	if len(parts) == 0 {
+		parts = []string{""}
+	}
+
+	chars := []rune(line)
+
+	if len(chars) != 0 && unicode.IsSpace(chars[len(chars)-1]) {
+		parts = append(parts, "")
+	}
+
+	lastTokIndx := len(parts) - 1
+	lastTok := parts[lastTokIndx]
+
+	if lastTokIndx == 0 {
+		return r.completeCmd(lastTok)
+	}
+
+	var compls []string
+	// This isn't ideal, but the directives are baked into tags
+	// making this trickier to implement neatly
+	switch parts[0] {
+	case ":load":
+		compls = r.completeFile(lastTok)
+	case ":let":
+		compls = r.completeVar(lastTok)
+	case ":exec":
+		compls = r.completeRule(lastTok)
+	}
+
+	res := make([]string, len(compls))
+	for i, c := range compls {
+		parts[lastTokIndx] = c
+		res[i] = strings.Join(parts, " ")
+	}
+
+	return res
+}
+
+func (r *REPL) completeCmd(prefix string) (c []string) {
+	var matches []string
+
+	ds := []string{
```

Comment:
This array can be a package-level variable to avoid allocating it over and over again whenever `completeCmd` is called.